### PR TITLE
TRUNK-6203: Global properties access should be privileged

### PR DESCRIPTION
**Pull request by wikumChamith**
## Description of what I changed

After https://github.com/openmrs/openmrs-core/pull/4601, users must possess the 'Get Global Properties' privilege to access global properties. This grants it as a proxy privilege so users can access essential pages, such as the login page.

## Issue I worked on

see https://openmrs.atlassian.net/browse/TRUNK-6203

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java [hunk 1]
```diff
@@ -210,8 +211,16 @@ public static boolean ipMatches(String ip, List<String> candidateIps) {
 	 * @return the list of IPs
 	 */
 	public static List<String> getAllowedIps() {
-		String allowedIpsProperty = Context.getAdministrationService()
-		        .getGlobalProperty(RestConstants.ALLOWED_IPS_GLOBAL_PROPERTY_NAME, "");
+		String allowedIpsProperty = "";
+		try {
+			Context.addProxyPrivilege("Get Global Properties");
```
Comment (dkayiwa) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java:216]: Why not use `PrivilegeConstants`?

Comment (wikumChamith) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java:216]: `PrivilegeConstants.GET_GLOBAL_PROPERTIES` is not available in the platform version that this module uses

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java [hunk 2]
```diff
@@ -210,8 +211,16 @@ public static boolean ipMatches(String ip, List<String> candidateIps) {
 	 * @return the list of IPs
 	 */
 	public static List<String> getAllowedIps() {
```
[the remark this answers is not in the document]
Comment (wikumChamith) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java:213]: This is called before accessing the O3 login page.

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java (hunk 1 above)
Comment (dkayiwa) [reply omod-common/src/main/java/org/openmrs/module/webservices/rest/web/RestUtil.java:216]: Which version is it?

**Comment by wikumChamith** [comment]
@dkayiwa I updated this PR.

**Pull request merged**